---
title: "Feli Capstone Project README.md"
output: html_document

---
#output: 
#  md_document:
#     variant: markdown_github
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```



```{r}
library(tidyverse)
library(readr)
```

https://github.com/fredhutchio/tfcb_capstone
https://www.markdownguide.org/cheat-sheet
##About the Data
This is where I talk about the data, assumptions etc

The dataset that I chose to investigate further was the [Immunohistochemical typing of adenocarcinomas](https://datadryad.org/stash/dataset/doi:10.5061/dryad.g8h71).
I am specifically working with the dataset that was found in the raw folder in "pcbil_clinicaldata.xlsx". This can be fund in my repository folder at  ["pcbil_clinicaldata.xlsx"](data/pcbil_clinicaldata.xlsx). Now I will specifically be working with data that is found at ["pcbil_clinicaldata.xlsx"](data/pcbil_clinicaldata.xlsx) but in the "PLOS Revision" tab. I have placed the data that was in the "PLOS Revision" tab into a new csv file found [here](data/pcbil_clinicaldata_PLOS_REVISION.csv). 


The paper delves into Adenocarcinomas of the pancreatobiliary system and how they are classified. This data comes from an observational cohort study on patients with Adenocarcinoma of the pancreatobiliary system. Certain morphological features of carcinomas make it difficult to distinguish between carcinomas. For instance intrahepatic cholangiocarcinoma is difficult to distinguish based on morphology so other factors such as immunhistochemistry may be used. 

"hepatocellular carcinoma is the control tumor type becuase immunohistochemistry is well-characterized"

## Project organization

Information about things in this project

Within the data folder, I have placed the dataset that I extracted from DRYAD. The dataset had a "pcbil_clinicaldata.xlsx" excelsheet but in order to make dtaaframes in Python, I made separate CSV files for each sheet in the excel workbook.

## Data

Citation and link to data

I will specifically be working with data that is found at ["pcbil_clinicaldata.xlsx"](data/pcbil_clinicaldata.xlsx) but in the "PLOS Revision" tab. I have placed the data that was in the "PLOS Revision" tab into a new csv file found [here](data/pcbil_clinicaldata_PLOS_REVISION.csv). 

One assumptions I made about the data was the last time it was assessed. This is an observationaly data set that followed patients until August 2016. One of my questions involved age and so I assumed if the date od death was not listed, then they were still alive at least until the last time the data was assessed. I'm not sure exactly what day in August 2016 they stopped being followed but I listed the "date of death" as the last day the data was assessed just so that I can calculate Age based on days. So I did change the cells from "NaN" to present so that I could calculate the age. This edited csv file can be found [here](data/pcbil_clinicaldata_PLOS_REVISION_edited.csv).

Since I am using the datset that has already been precoessed by the group, I will describe a little what pre-processing the group did. They had originally used 38 antibody markers but reduced it down to 27 due to missing data. These markers are used commonly for diagnosing pathology such as markers for mucins to be able to distinguish between squamous and glandular and so.
Additionally, they imputed the dataset with the Bioconductor package “impute”. My analysis will be on the imputed dataset.



## About the analysis

Information about other things useful to know for the context of the overall project.

#immunohistochemical markers comparison?
#probe used? type of probe (biopsy/resection
#further understand the differences in marker expression between the immunohistochemical tumor types

Proposal of diagnostic immunohistochemical panel for the classification of the immunohistochemical tumor types.
Extrahepatic pancreatobiliary,	Intestinal,	Intrahepatic cholangiocarcinoma,	Hepatocellular carcinoma

1. The authors noted that females with adenocarcinoma of the pancreato-biliary system had a lower median overall survival (months) than males (Table 2 in their paper). One question I wanted to ask was if females with Hepatocellular Cancer, so the control cancer, had a lower overall survival than males.

2. One question that I wanted ask

3. One fun question that I wanted to ask with R is: Is there a sex-based differences in identifying if a patient has intrahepatic cholangiocarcinoma? Additionally with this question, I wanted to see if there are also differences in care. For instance, are males more likely to receive chemo than females in this subset?

##Reproducibility




## Problem 1

**10 points**

For each of the following functions, provide a <100 character description
(in your own words) and a URL reference.

1. `!`
This mean "not". For example, when using the logical operator "!=" this would mean "not equal to" [.](https://www.statmethods.net/management/operators.html)

2. `is.na`
This is a function that can be used to search for missing values in a dataset, returning true if NA [present.](https://www.programmingr.com/tutorial/is-na/#:~:text=Find%20missing%20values%20in%20R,to%20a%20value%20of%20false.)

3. `is.numeric`
This function test returns 'True' if all numbers in the dataset or object tested are [numeric.](https://www.rdocumentation.org/packages/base/versions/3.3/topics/numeric) 

4. `anti_join`
This takes two arguments (let's say x/y) and returns all rows from x where it does not match values in  [y.](http://zevross.com/blog/2014/08/05/using-the-r-function-anti_join-to-find-unmatched-records/)

5. `desc`
This is a function that takes the vector as the input and transforms it so that the data can be sorted in descending [order.](https://www.rdocumentation.org/packages/plyr/versions/1.8.6/topics/desc)

6. `slice`
This is a function that will only depict rows of interest from some dataset. For instance: data %>% 
  slice (1:10) [.](https://dplyr.tidyverse.org/articles/base.html?q=slice#slice-choose-rows-by-position)

7. `all_vars`
This is a function that is used often with filter() functions to return variables that are True and meet the condition [ex) all_vars(. >= 100) ](https://dplyr.tidyverse.org/reference/all_vars.html)

8. `funs`
Funs is a function that creates a lit of functions. So you can create a list that contains only the mean, max, and median functions [. ](https://www.rdocumentation.org/packages/dplyr/versions/0.7.8/topics/funs)

9. `filter_if`
This is a filtering function that will filter for variables/rows/etc that meet a specific [condition.](http://thatdatatho.com/2019/04/08/more-dplyr-get-data-into-right-shape/)

10. `mutate_if`
This function is used to transform/affect multiple variables that meet a specific  [condition.](http://thatdatatho.com/2019/04/08/more-dplyr-get-data-into-right-shape/)  


## Problem 2

**10 points**

Add a comment above each code line below explaining what the code line does and/or why
that code line is necessary.

Keep each comment to less than 2 lines per line of code and < 80 chars per line.

```{r}

# You are reading in the tsv file into a tibble assigned as 'annotations'.
annotations <- read_tsv("ftp://ftp.ebi.ac.uk/pub/databases/genenames/new/tsv/locus_groups/protein-coding_gene.txt") %>%
  #Using pipe %>% operator to chain commands. Select() command will only keep variables mentioned.
  select(ensembl_gene_id, symbol, name, gene_family, ccds_id) %>%
  #Further pipe `annotations` to filter for all data except instances where the ccds_id is not known.
  filter(!is.na(ccds_id)) %>%
  #Now you are printing what you have selected and filtered with the chained commands. 
  print()


```

```{r}
# You are reading in the tsv file into a tibble assigned as 'data'.Pipe operator to chain commands
data <- read_tsv("ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE89nnn/GSE89183/suppl/GSE89183_Counts.txt.gz") %>%
  #Chained command is rename() to change column name. 
  rename(ensembl_gene_id = "ENSEMBL gene") %>%
  #Now printing all rows of 'data' dataset
  print()
#write_csv(data, "hw_6_file.tsv") 
```

## Problem 3

**10 points**

Using the code below:

#1. Convert both axes to `log10` instead of linear scales.
#2. Show axis tick labels as 10^0^, 10^1^, 10^2^,10^3^, 10^4^, 10^5^ for both axes.
#3. There are too many points overlapping in certain regions. Use a different
`geom_` function to convey to your reader how many overlapping points are present
in each region.


```{r, fig.width=4, fig.height=4}
data %>%
  select(CD34_shRPL5_RNA_1, CD34_shRPS19_RNA_1) %>%
  ggplot(aes(x = CD34_shRPL5_RNA_1, y = CD34_shRPS19_RNA_1)) +
  #geom_hex(binwidth = c(.1, 500), stat = "binhex",) +
  geom_bin2d(binwidth = c(0.1, 0.1)) +
  scale_x_log10(
    breaks= c(1, 10, 100, 1000, 10000, 100000), 
    labels= c("10^0", "10^1", "10^2", "10^3", "10^4", "10^5")) +
  scale_y_log10(
    breaks= c(1, 10, 100, 1000, 10000, 100000), 
    labels= c("10^0", "10^1", "10^2", "10^3", "10^4", "10^5")) #+
  #labs(x = "Kozak region", y= "Mean ratio", title= "Effect of Kozak region on protein expression")

#"10^0", "10^1", "10^2", "10^3", "10^4", "10^5"
#geom_count()
```

**In problems 4 through 6, assign the result of your operation back to the `data`
variable.**

## Problem 4

**10 points**

Write a code chunk to select the following columns
from the `data` variable you created above
and reassign back to `data`.
```{r}
data <-data %>%
  select(ensembl_gene_id, CD34_shRPL5_RPF_1, CD34_shRPL5_RPF_2, CD34_shRPL5_RNA_1, CD34_shRPL5_RNA_2,
         CD34_shRPS19_RPF_1, CD34_shRPS19_RPF_2, CD34_shRPS19_RNA_1, CD34_shRPS19_RNA_2, CD34_shLuc_RPF_1,
         CD34_shLuc_RPF_2, CD34_shLuc_RNA_1, CD34_shLuc_RNA_2)
  
```



## Problem 5

**10 points**

Write a code chunk to filter the result from Problem 4 to include
only rows where each of the 12 numerical columns
you selected has 50 counts or more  and reassign back to `data`.
This is a simple way to avoid genes that have
very low counts. You might be tempted to do this step separately for
each of the 12 columns, but
you will receive 5 bonus points if you instead use the `filter_if`
and `all_vars` functions you learned above.
```{r}
data <-data %>%
  filter_if(is.numeric, all_vars(. >= 50)) %>%
  print()

#write_csv(data, "hw_6_file2.tsv") 

```

## Problem 6


**10 points**

Write a code chunk to divide each of the 12 numerical columns by the
corresponding median value for each column
and reassign back to `data`. This median normalization is typically done in
high-throughput experiments after filtering
to normalize for sample-to-sample difference in read depth.
Again, you can write lot less code if you use the `mutate_if` and `funs` functions
you learned above.

```{r}

data <-data %>%
  #filter_if(is.numeric, all_vars(. >= 50)) %>%
  mutate_if(is.numeric, funs(./median(.))) %>%
  print()

```
  # mutate( CD34_shRPL5_RPF_1 = CD34_shRPL5_RPF_1/median(CD34_shRPL5_RPF_1),
  #         CD34_shRPL5_RPF_2=CD34_shRPL5_RPF_2/median(CD34_shRPL5_RPF_2),
  #         CD34_shRPL5_RNA_1=CD34_shRPL5_RNA_1/median(CD34_shRPL5_RNA_1),
  #         CD34_shRPL5_RNA_2=CD34_shRPL5_RNA_2/median(CD34_shRPL5_RNA_2),
  #         CD34_shRPS19_RPF_1=CD34_shRPS19_RPF_1/median(CD34_shRPS19_RPF_1), 
  #         CD34_shRPS19_RPF_2=CD34_shRPS19_RPF_2/median(CD34_shRPS19_RPF_2), 
  #         CD34_shRPS19_RNA_1=CD34_shRPS19_RNA_1/median(CD34_shRPS19_RNA_1), 
  #         CD34_shRPS19_RNA_2= CD34_shRPS19_RNA_2/median(CD34_shRPS19_RNA_2), 
  #         CD34_shLuc_RPF_1=CD34_shLuc_RPF_1/median(CD34_shLuc_RPF_1),
  #         CD34_shLuc_RPF_2=CD34_shLuc_RPF_2/median(CD34_shLuc_RPF_2),
  #         CD34_shLuc_RNA_1=CD34_shLuc_RNA_1/median(CD34_shLuc_RNA_1),
  #         CD34_shLuc_RNA_2=CD34_shLuc_RNA_2/median(CD34_shLuc_RNA_2)
  #         )%>%

## Problem 7 

**10 points**

After we do the above filtering and median-normalization, let us calculate
translation efficiency as the average ratio of the RPF and RNA reads for each
treatment condition. Then we calculate how this translation efficiency changes
between target (`rpl5` and `rps19`) and control (`luc`) shRNAs.

The code implementing the above steps is shown below, but it has a few
errors. Correct them.

```{r}
library(tidyverse)
library(tibble)
library(tidyr)
library(readr)
library(ggplot2)
lfc <- data %>%
  mutate(mean_rpl5_te = ((CD34_shRPL5_RPF_1 + CD34_shRPL5_RPF_2) /
                            (CD34_shRPL5_RNA_1 + CD34_shRPL5_RNA_2)) )%>%
  mutate(mean_rps19_te = ((CD34_shRPS19_RPF_1 + CD34_shRPS19_RPF_2) /
                            (CD34_shRPS19_RNA_1 + CD34_shRPS19_RNA_2)) )%>%
  mutate(mean_shluc_te = ((CD34_shLuc_RPF_1 + CD34_shLuc_RPF_2) /
                            (CD34_shLuc_RNA_1 + CD34_shLuc_RNA_2))) %>%

  select(ensembl_gene_id, mean_rpl5_te, mean_rps19_te, mean_shluc_te) %>%
  mutate(lfc_te_rpl5 = log2(mean_rpl5_te / mean_shluc_te),
         lfc_te_rps19 = log2(mean_rps19_te / mean_shluc_te)) %>%
  print()
#write_csv(lfc, "hw_6_file_problem7.tsv") 
```



## Problem 8 

**10 points**

Write code that will create a new dataframe called `mean_lfc` from `lfc`
containing a new column called `avg_lfc`.
`avg_lfc` should be the average of the log2 fold-change in TE (`lfc_te`) upon
knockdown of RPL5 and RPS19.

Then select only the gene id column and the new column that you just created
(this will be your new dataframe `mean_lfc`).
```{r}
mean_lfc <-lfc %>%
  mutate(avg_lfc =((lfc_te_rpl5+lfc_te_rps19)/2) ) %>%
  select(ensembl_gene_id, avg_lfc ) %>%
  print()
  

```


## Problem 9

**10 points**

Write code to join the `mean_lfc` dataframe with the `annotations` dataframe created
at the top of the document and assign back to `mean_lfc`.
```{r}
mean_lfc <- mean_lfc %>%
  left_join(annotations, by = "ensembl_gene_id") %>% 
  print()
#or full_join()
```




## Problem 10 

**10 points**

1. Write code to select only the bottom 10 genes with the
lowest `avg_lfc` and display the
gene `symbol`, gene `name` and `avg_lfc` for these genes.
2. Create a figure using ggplot to visualize these results.
Write a few sentences to justify the choices you made when creating your figure.
```{r}
mean_lfc %>%
  #group_by(ensembl_gene_id) %>%
  select("name", "symbol", "avg_lfc")%>%
  arrange(avg_lfc) %>%
  slice(0:10)%>%
  print()%>%
  ggplot(aes(x = symbol, y = avg_lfc, label= round(avg_lfc, 2) )) +
  geom_point() +
  geom_text( hjust = 0, nudge_y = 0.1, nudge_x = -0.1,  size=5)+
  theme_classic()+
  scale_y_continuous(limit= c(-3,-1)) +
  
  labs(x = "Gene symbol", y= "Average of the log2 fold-change in TE", size=5)
 #print("name", "symbol", "avg_lfc" )

'Explanation of figure: I chose to use `theme_classic()` becuase I wanted to remove the grey background and just have a white background. I adjusted the y-axis with `scale_y_continuous` in order to  make the position of the data slightly more visually appealing. I used `geom_text` to label each data point with the gene symbol. I modified the label so the text is slightly to the right of the data point. I chose to use `geom_point` because `geom_hist` might have had overlap. With `geom_point`, I could show the discrete data points. '
  
  #filter(min_rank(avg_lfc)<= 10)
```




**When you are satisfied with your code and answers,
use the "Knit" button in RStudio to create the final set of files.
Save the result as a PDF and  and submit it.**
